README.md

```diff
@@ -1,171 +1,577 @@
 
-To avoid slowing down the seed process due to multiple database calls we recommend that you initialise a subset of data into the DbSeed from which you can then randomise the results.
+#### Set a property on a parent entity:
+```csharp
+new BookingSeed().With(m => m.Member.FirstName, "Member Name");
+```
+#### Set a navigation property:
+```csharp
+new BookingSeed().With(m => m.Member, new Member() {...});
+```
 
-In the example below, the database will only ever be called on the first usage of `GetSingle()` when configuring seeds, future calls will reuse the Products array as DbSeeds are persisted until all seed data has been processed.
+### `With` (multiple entities)
+The following examples apply to when you're seeding multiple entities (i.e with `SeedMany`).
 
+#### Set a property to a single value for all entities:
 ```csharp
-    public class OrderSeed : SeedFromDatabase<Order>, IDependsOn<Product>
-    {
-        private Product[] Products { get; set; }
+new BookingSeed().With(b => b.Name, "Booking Name");
+```
+> [!NOTE]
+> In this scenario, all entities will have the same value for the property.
 
-        public override Order GetSingle()
-        {
-            // Intialise a list of products from the database when null
-            // This list will be persisted until the seed process has completed
-            Products ??= DbEntity<Product>()
-                .Where(p => p.Deleted == null)
-                .Where(p => p.Price > 100)
-                .ToArray();
+#### Set a property to multiple values in order:
+```csharp
+new BookingSeed().With(b => b.Name, "Booking A", "Booking B", "Booking C");
+```
+
+#### Set a property based on the index of the entity being seeded
+```csharp
+new CouponSeed().With(c => c.Discount, index => index * 10m);
+```
+
+#### Set a property based on the previous entity:
+```csharp
+new CouponSeed().With(c => c.Discount, (_, previous) => previous?.Discount * 2 ?? 0.1m)
+```
 
-            var order = new Order();
+#### Set a property with a new value each time:
+```csharp
+new CouponSeed().With(
+    scs => scs.Name,
+    // Delegate is re-evaluated each time
+    () => Guid.NewGuid().ToString())
+```
+### `Without`
 
-            // Get 10 random products and add to order
-            var randomProducts = Random.Elements(Products, 10);
-            order.AddRange(randomProducts);
+#### Set a property on an entity as `null`;
+```csharp
+new BookingSeed().Without(b => b.Name);
+```
 
-            return order;
-        }
-    }
+
+### `WithNew` (single entity)
+Specify that an navigation optional property should be non-null, or override the default behaviour for a mandatory navigation property.
+#### Set a navigation property without a seed:
+```csharp
+new BookingSeed().WithNew(b => b.Coupon);
+```
+In this scenario, we will look for inheritors of `EntitySeed<Coupon>`, falling back on the default behaviour if none are found (see [Seeding without a class](#seeding-without-a-class)).
```

Comment:
Link doesn't work - there is no "Seeding without a class" heading.

Reply:
thanks - this is now "reflection-based-seeding"